lib/sycamore/sycamore/tests/integration/connectors/opensearch/test_opensearch_read.py

```diff
@@ -110,3 +115,89 @@ def test_ingest_and_read(self, setup_index, exec_mode):
 
         for i in range(len(doc.elements) - 1):
             assert doc.elements[i].element_index < doc.elements[i + 1].element_index
+
+    def _test_ingest_and_read_via_docid_reconstructor(self, setup_index, exec_mode, cache_dir):
+        """
+        Validates data is readable from OpenSearch, and that we can rebuild processed Sycamore documents.
+        """
+
+        print(f"Using cache dir: {cache_dir}")
+
+        def doc_reconstructor(index_name: str, doc_id: str) -> Document:
+            prefix = f"doc-{doc_id}"
+            found = None
+            files = os.listdir(cache_dir)
+            for f in files:
+                if f.startswith(prefix):
+                    found = f
+                    break
+            assert found, "Doc not found in cache"
+            import pickle
+
+            data = pickle.load(open(f"{cache_dir}/{found}", "rb"))
+            return Document(**data)
+
+        path = str(TEST_DIR / "resources/data/pdfs/Ray.pdf")
```

Comment:
I would recommend fake documents rather than running the partitioner. This test is testing way more than is necessary.  As a side benefit, it becomes very easy to verify that you're reconstructing via pickle. The fake documents can do doc.hidden_var = <something>; and then you check that hidden_var pops out on the far side.  Since it wouldn't be written to opensearch, you know it's doing pickle reconstruction.

Also it seems better to test with multiple documents which is easier if they're fake.